src/include/rados/rgw_file.h

```diff
@@ -221,7 +221,7 @@ int rgw_unlink(struct rgw_fs *rgw_fs,
 /*
     read  directory content
 */
-typedef bool (*rgw_readdir_cb)(const char *name, void *arg, uint64_t offset,
+typedef int (*rgw_readdir_cb)(const char *name, void *arg, uint64_t offset,
```

Comment:
does cython not have a bool type?

Reply:
python boolean types are converted to "int" to "bint" not "bool".
see: https://cython.readthedocs.io/en/stable/src/userguide/language_basics.html#types

Reply:
I don't really care much, yet I really don't feel like cython's interface should be driving changes rgw_file's type signatures...

Reply:
Maybe it's *slightly* relevant to note that bool is a neologism--or was one, 22 years ago when it was standardized

Reply:
> I don't really care much, yet I really don't feel like cython's interface should be driving changes rgw_file's type signatures...

i don't mind changing the cython file instead

Reply:
@mattbenjamin i don't see another way to workaround that. without that fix, we won't be able to build with the latest clang (16).

Reply:
just another python fail, I guess